### PR TITLE
examples: add ci logic, fix import, run tests

### PR DESCRIPTION
**Pull request by dragonsinth**
We should make sure that examples work.

### CODE REVIEW

File: examples/storage/example_test.go [hunk 1]
```diff
@@ -4,12 +4,12 @@ import (
 	"context"
 	"fmt"
 	"io"
-	"io/ioutil"
 	"os"
 	"strings"
 	"testing"
 
 	"github.com/fullstorydev/emulators/storage/gcsemu"
+	testcontainers "github.com/testcontainers/testcontainers-go"
```
Comment (dragonsinth) [inline examples/storage/example_test.go:12]: this import actually got lost in a previous commit

File: examples/go.mod [hunk 2]
```diff
@@ -1,6 +1,8 @@
 module github.com/fullstorydev/emulators/examples
 
-go 1.18
+go 1.21
+
+toolchain go1.21.6
```
Comment (gpassini) [inline examples/go.mod:5]: Must we enforce a Go version higher than the one required by the actual emulators packages?
Currently they require only `go 1.18`.

Comment (gpassini) [reply examples/go.mod:5]: Actually this seems to be what's breaking CI for Go versions < 1.21.

Comment (dragonsinth) [reply examples/go.mod:5]: oh!  that's annoying.. it just did that automatically :/
I'll revert.

File: examples/Makefile [hunk 3]
```diff
@@ -0,0 +1,53 @@
+.PHONY: ci
+ci: deps checkgofmt vet staticcheck ineffassign predeclared golint errcheck test
+
+.PHONY: deps
+deps:
+	go get -d -v -t ./...
```
Comment (dragonsinth) [inline examples/Makefile:6]: Something about this upgrades the go.mod automatically, but I'm not sure why.  The one in bigtable (for example) is happy to stay on 1.18 😦

Comment (dragonsinth) [reply examples/Makefile:6]: Ah ok, I have to roll back testcontainers version to one that supports go 1.18

**Comment by dragonsinth** [comment]
I'm gonna be real, I'm pretty much not a fan of the whole testcontainers thing and pretty inclined to just rip it out.  But for now I'll try only compiling that on 1.21.

**Comment by dragonsinth** [comment]
@franklinlindemberg do you want to lobby for keeping the testcontainers test code?

**Comment by franklinlindemberg** [comment]
> @franklinlindemberg do you want to lobby for keeping the testcontainers test code?

nope! Im fine to remove it

File: examples/go.mod [hunk 4]
```diff
@@ -2,11 +2,12 @@ module github.com/fullstorydev/emulators/examples
 
 go 1.18
 
+toolchain go1.21.6
```
Comment (gpassini) [inline examples/go.mod:5]: 🤔 The `toolchain` is still being automatically added?

Comment (dragonsinth) [reply examples/go.mod]: No, I think I messed up.  Fifth time's a charm?

**Review by gpassini — APPROVED** [review]
LGTM!

**Pull request merged**